includes/base/element-base.php

```diff
@@ -523,6 +542,30 @@ public function print_element() {
 		do_action( 'elementor/frontend/after_render', $this );
 	}
 
+	protected function should_render_shortcode() {
+		$should_render_shortcode = apply_filters( 'elementor/element/should_render_shortcode', false );
+
+		if ( ! $should_render_shortcode ) {
+			return false;
+		}
+
+		$raw_data = $this->get_raw_data();
+
+		if ( ! empty( $raw_data['settings']['_element_cache'] ) ) {
+			return 'yes' === $raw_data['settings']['_element_cache'];
+		}
+
+		if ( $this->is_dynamic_content() ) {
+			return true;
+		}
+
+		if ( empty( $raw_data['settings']['__dynamic__'] ) && empty( $raw_data['settings']['e_display_conditions'] ) ) {
```

Comment:
Is it ok to have such display conditions reference in core?

Reply:
I agree. The Display conditions module should implement the `elementor/element/should_render_shortcode` filter.